packages/react/src/components/IconButton/index.tsx

```diff
@@ -5,16 +5,96 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import PropTypes from 'prop-types';
-import React from 'react';
+import PropTypes, { ReactNodeLike } from 'prop-types';
+import React, { ForwardedRef } from 'react';
 import Button from '../Button';
 import classNames from 'classnames';
 import { Tooltip } from '../Tooltip';
 import { usePrefix } from '../../internal/usePrefix';
 import cx from 'classnames';
 
-const IconButton = React.forwardRef(function IconButton(props, ref) {
-  const {
+interface IconButtonProps
+  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  /**
+   * Specify how the trigger should align with the tooltip
+   */
+  align?:
+    | 'top'
+    | 'top-left'
+    | 'top-right'
+    | 'bottom'
+    | 'bottom-left'
+    | 'bottom-right'
+    | 'left'
+    | 'right';
+
+  /**
+   * Provide an icon or asset to be rendered inside of the IconButton
+   */
+  children?: React.ReactNode;
+
+  /**
+   * Specify an optional className to be added to your Button
+   */
+  className?: string;
+
+  /**
+   * Determines whether the tooltip should close when inner content is activated (click, Enter or Space)
+   */
+  closeOnActivation?: boolean;
+
+  /**
+   * Specify whether the tooltip should be open when it first renders
+   */
+  defaultOpen?: boolean;
+
+  /**
+   * Specify whether the Button should be disabled, or not
+   */
+  disabled?: boolean;
+
+  /**
+   * Specify the duration in milliseconds to delay before displaying the tooltip
+   */
+  enterDelayMs?: number;
+
+  /**
+   * Specify whether the IconButton is currently selected
+   */
+
+  isSelected?: boolean;
+
+  /**
+   * Specify the type of button to be used as the base for the IconButton
+   */
+  kind?: 'primary' | 'secondary' | 'ghost' | 'tertiary';
```

Comment:
Seems like there is a type mismatch here since we don't allow danger icon buttons, but `danger` is a valid `kind` for `Button`

Reply:
I just updated the types to use the formal Button `kind` types, and omit the danger variants that we don't allow for IconButtons

Reply:
Hey I just took another look and this mismatch persists even when using Omit on the ButtonKind type. I'll look into it.